qdrant-landing/content/articles/what-is-quantization.md

```diff
@@ -0,0 +1,242 @@
+
+```python 
+collection_config = {
+    "vectors": {
+        "size": 1024,
+        "distance": "Cosine"
+    },
+    "quantization_config": {
+        "product": {
+            "compression": "x32",
+            "always_ram": True
+        }
+    }
+}
+```
+
+Each region in the codebook is defined by a **centroid**, which serves as a representative point that summarizes the characteristics of that region. So, instead of treating every single data point as equally important, we can group similar points together and represent them with a single centroid that is more representative of the general characteristics of that group.
+
+The centroids used in Product Quantization are determined using the **[K-means clustering algorithm](https://en.wikipedia.org/wiki/K-means_clustering)**, which Qdrant applies to a representative sample of the dataset you want to quantize.
+
+
+![Codebook and Centroids example](/articles_data/what-is-vector-quantization/codebook.png)
+
+
+Qdrant always selects **K = 256** for the number of centroids in its implementation based on the fact that 256 is the maximum number of unique values that can be represented by a single byte.
+
+This makes the compression process efficient because each centroid index can be stored in a single byte.
+
+After the codebook is created, the original high-dimensional vectors are split into smaller **sub-vectors.** 
+
+Each sub-vector is then compared to every centroid in the codebook and mapped to the nearest centroid.
+
+![Vectors being mapped to their correspondent centroids example](/articles_data/what-is-vector-quantization/centroids-mapping.png)
+
+The compressed vector stores the index of the closest centroid for each subvector.
+
+Here’s how a 1024-dimensional vector originally taking up 4096 bytes is reduced to just 128 bytes by representing it as 128 indexes, each pointing to the centroid of a subvector:
+
+
+![Product Quantization example](/articles_data/what-is-vector-quantization/product-quantization.png)
+
+After setting up quantization and adding your vectors, you can perform searches as usual. Qdrant will automatically use the quantized vectors, optimizing both speed and memory usage. Optionally, you can enable rescoring for better accuracy.
+
+```python 
+search_payload = {
+    "query": [0.22, -0.01, -0.98, 0.37, ...],
+    "params": {
+        "quantization": {
+            "rescore": True
+        }
+    },
+    "limit": 10
+}
+```
+
+Product Quantization is generally more memory efficient than **Scalar Quantization**, being able to reduce the memory footprint by up to **64x** while still retaining much of the essential information. However, the computational overhead of mapping sub-vectors to centroids results in the **slowest performance** compared to the other methods.
```

Comment:
Should we describe this method to retain much of the essential information, especially if we mention the 64x config in the same sentence? Overall, the quality drops quite significantly in our benchmarks:

![image](https://github.com/user-attachments/assets/593ed92f-df55-4209-a4f8-4f028ab6ee95)

Also, PQ always results in slower indexing, but the search might be even faster than with the original vectors:

![image](https://github.com/user-attachments/assets/ff28a1c9-a839-4eba-ac44-8371d207eb12)

Reply:
In our benchmarks, what's the compression we used? I think 0.7 precision would be for 64x, correct? I'd assume for smaller compression the loss would be less significant. If you have a notebook I can use for some benchmarks that'd be great

Reply:
Also, can you clarify on why is the speed 0.5? Is that including the indexing?